### PR TITLE
Introduce automated tests

### PR DESCRIPTION
**Pull request by howard-e**
[Preview Tests](https://deploy-preview-1105--aria-at.netlify.app)

This introduces tests which compare the snapshots of a 4 specific test plans which have been pulled into a dedicated `__mocks__` folder:
* alert (using v2 test format version)
* banner (using v1 test format version)
* command-button (using v2 test format version, with more complexity than alert)
* horizontal-slider (using v2 test format version, which includes assertionExceptions)

This PR does significant restructuring of the `create-all-tests` and the `review-tests` scripts to make the results of those overall scripts testable (ie. content that gets added into the `/build` folder).

Additional tests can follow this PR to verify the many utilities and the harness output used in this project. To avoid ballooning this PR further, we can introduce this initial work and do the aforementioned tests as subsequent PRs to make future reviews easier.

Note: Also confirmed these changes don't cause any critical issues when being [imported into aria-at-app](https://github.com/w3c/aria-at-app/compare/test-ariaat-1105). [(build log)](https://github.com/w3c/aria-at-app/actions/runs/10407221729/job/28822017572)

### CODE REVIEW

File: .github/workflows/generate-and-commit-files.yml [hunk 1]
```diff
@@ -5,7 +5,12 @@ on:
 
 jobs:
   generate-and-commit-files:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
Comment (stalgiag) [inline .github/workflows/generate-and-commit-files.yml:8]: Nice!

**Review by stalgiag — APPROVED** [review]
Read through all of the changes and everything looks good! Also ran the test and love this new, easier entry point for interacting with the repo. The only two comments that I have are both non-blocking, nice-to-haves.

1. `createReviewPages.mjs` feels like it could be broken up a little to make it easier to read. Some sections become deeply nested.
2. The logging during testing is difficult to parse. I wonder if this could be helped with something simple like [an emphasis color](https://davidlozzi.com/2021/03/16/style-up-your-console-logs/) on the summary messages? This isn't essential but could be nice.

**Comment by howard-e** [comment]
> 1. `createReviewPages.mjs` feels like it could be broken up a little to make it easier to read. Some sections become deeply nested.
> 2. The logging during testing is difficult to parse. I wonder if this could be helped with something simple like [an emphasis color](https://davidlozzi.com/2021/03/16/style-up-your-console-logs/) on the summary messages? This isn't essential but could be nice.

Good point on both, will take a pass at making that content easier to parse

File: scripts/test-reviewer/createReviewPages.mjs [hunk 2]
```diff
@@ -0,0 +1,600 @@
...
+                commandsValuesForInstructions = assertionsForCommandsInstructions;
+                assertionsForCommandsInstructions = assertionsForCommandsInstructions.map(each => ({
+                  value: each,
+                }));
+              }
+            }
+
+            // For V2 to handle assertion exceptions
+            assertionsForCommandsInstructions = assertionsForCommandsInstructions.map(
+              (assertionForCommand, assertionForCommandIndex) => {
```
[the remark this answers is not in the document]
Comment (howard-e) [reply scripts/test-reviewer/createReviewPages.mjs:337]: Will do!

**Comment by howard-e** [comment]
> 1. `createReviewPages.mjs` feels like it could be broken up a little to make it easier to read. Some sections become deeply nested.

@stalgiag I added several commits to break up the `createReviewPages.mjs` into several smaller components. Let me know me know your thoughts there!

> 2. The logging during testing is difficult to parse. I wonder if this could be helped with something simple like [an emphasis color](https://davidlozzi.com/2021/03/16/style-up-your-console-logs/) on the summary messages? This isn't essential but could be nice.

https://github.com/w3c/aria-at/pull/1105/commits/b33362270bd9f7684a37f9c45415c9bdb8a05c5c should address this. Please let me know if this aligns with what you were thinking. Also curious if you think this functionality should be set behind the test flag?

**Review by stalgiag — APPROVED** [review]
Wow you went above and beyond! Thanks for addressing the feedback. The new `createReviewPages` organization is much easier for me to follow with smart separation of responsibilities.

The log styling is great and I like it as-is. With scripts that produce lots of output, like this one, it is useful to err on the side of conveying more contextual/severity info where possible.

**Comment by howard-e** [comment]
@stalgiag thanks for the feedback! Found a presumably long standing issue (but presentational) with how the `assertionExceptions` logic is handled in when built with the json data that I've addressed in https://github.com/w3c/aria-at/pull/1105/commits/c66455fec3847d9e7ad887848fe0676663023988. The gist being that [json filtered view](https://aria-at.netlify.app/tests/horizontal-slider/test-05-navforwardstoslider?at=jaws) could show all the `assertionExceptions` if excluded in `assertions.csv` instead of `*-commands.csv`. The [collected json view (which is also what the app uses)](https://aria-at.netlify.app/tests/horizontal-slider/test-05-navforwardstoslider-jaws.collected.html) didn't have this issue.

<details>
<summary>screenshot showing json filtered view incorrectly showing 'jaws switched from virtual cursor active to pc cursor active' for 'down down' command</summary>

json filtered view incorrectly showing 'jaws switched from virtual cursor active to pc cursor active' for 'down down' command

</details>

<details>
<summary>screenshot showing collected json view correctly excluding 'jaws switched from virtual cursor active to pc cursor active' for 'down down' command</summary>

collected json view correctly excluding 'jaws switched from virtual cursor active to pc cursor active' for 'down down' command

</details>

Also re-confirmed these changes don't cause any critical issues when being [imported into aria-at-app](https://github.com/w3c/aria-at-app/compare/test-ariaat-1105). [(build log)](https://github.com/w3c/aria-at-app/actions/runs/10490554910/job/29057550927)

**Pull request merged**